pkg/resource/v3/resource.go

```diff
@@ -8,6 +8,8 @@ import (
 	hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
 )
 
+type Type string
+
```

Comment:
Nit: WDYT about making all the FooType constants below be of type `Type`? Too intrusive?

Reply:
Yea unfortunately that was intrusive and what caused all the code changes originally.